Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Dashboard/IssueList.hs

```diff
@@ -101,3 +103,16 @@ buildIssueList (issues, person) = do
         createdAt = issues.createdAt,
         updatedAt = issues.updatedAt
       }
+
+ticketStatusCallBack :: ShortId DM.Merchant -> Common.TicketStatusCallBackReq -> Flow APISuccess
+ticketStatusCallBack _ req = do
+  isPresent <- QIssue.isTicketPresent req.ticketId
+  unless isPresent $ throwError $ TicketDoesNotExist req.ticketId
+  runTransaction $ QIssue.updateIssueStatus req.ticketId (toDomainIssueStatus req.status)
```

Comment:
You may also use upsert by having a unique constraint on ticketId

Reply:
we are not creating status it always exists but this can be done in updation of ticketId for sure

Reply:
@0utkarsh are you sure `upsert` applicable here? Because here we do not `insert` any entity, only `update` and throw error if entity is not there. We should use `upsert` when we need both `insert` and `update` queries